kernel/kernel-api/src/main/java/io/delta/kernel/internal/TransactionImpl.java

```diff
@@ -228,14 +232,15 @@ public Optional<SetTransaction> getSetTxnOpt() {
     private Optional<Long> generateInCommitTimestampForFirstCommitAttempt(
```

Comment:
Can you add a docstring to indicate that this will automatically ensure that the returned timestamp is more than the one in the read snapshot?